### PR TITLE
chore: update to latest version 13 of nx and angular

### PR DESCRIPTION
**Pull request by Jefiozie**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {
   CodeGen,
   CodeGenState,
 } from '../../../core/state/page.state';
-import { codeGenList, UICodeGen } from '@xlayers-apps/shared/codegen-list';
+import { codeGenList, UICodeGen } from '../../../../shared/codegen-list';
```
Comment (manekinekko) [inline apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:10]: Is there is a reason we are not using TS aliases anymore?

File: libs/web-component-codegen/src/lib/web-component-codegen.service.spec.ts [hunk 2]
```diff
@@ -43,27 +43,27 @@ describe('WebComponentCodeGenService', () => {
     expect(actual).toBe('a_context');
   });
 
-  VERSION_LIST.forEach((version) => {
-    const fileNames = readdirSync(`${SKETCH_PATH}/${version}`);
+  // VERSION_LIST.forEach((version) => {
```
Comment (manekinekko) [inline libs/web-component-codegen/src/lib/web-component-codegen.service.spec.ts]: why are we commenting this test?

**Comment by Jefiozie** [comment]
Thank you for the review, I forgot to make it a draft. But I will comment on your question when I got everything working again. Need to resolve some issues here and there.

**Comment by Jefiozie** [comment]
@manekinekko, I've made all the changes necessary for the v13 update. However I can't figure out why the cypress test on a Mac fails. On my local machine (windows) everything runs as it should. Could you run it and see your result?


To answer your question regarding the type alias, the compiler couldn't resolve the aliases. I couldn't find a real reason for this other then that there are some changes in the NX part. NX these days have buildable libraries which makes the caching and building quicker because of parti compilation. We (xlayers) can benefit from this and I'm hoping to continue on this topic after this PR. Mainly for two reasons. 

1. Make the ci part quicker
2. Have a good demo/ viewable OSS repo where this is used. (Of course with your approval)

Also I took the liberty to upgrade some other packages to the latest versions.

**Comment by manekinekko** [comment]
Thank you for your amazing work! I will check the broken tests asap and keep you posted.

> Have a good demo/ viewable OSS repo where this is used. (Of course with your approval)

Please go ahead! I'd love to see that as well 😃

**Comment by Jefiozie** [comment]
Thank you for looking at the issue, when this PR is merged I will start with the buildable libs.

**Pull request closed without merging**